pkg/certificates/config_test.go

```diff
@@ -0,0 +1,433 @@
+package certificates
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func contains(s, substr string) bool {
+	return strings.Contains(s, substr)
+}
+
+// TestParseConfigFileNotFound tests the ParseConfig function with a non-existent file.
+func TestParseConfigFileNotFound(t *testing.T) {
+	_, err := ParseConfig("non-existent-file.yaml")
+	if err == nil {
+		t.Error("expected error for non-existent file but got none")
+	}
+	if !strings.Contains(err.Error(), "reading config file") {
+		t.Errorf("expected error message to contain 'reading config file', got %q", err.Error())
+	}
+}
+
+// setupSSHKeyForTest creates a temporary SSH key file for testing.
+func setupSSHKeyForTest(t *testing.T, path string) func() {
+	if path == "" {
+		return func() {}
+	}
+
+	if err := os.WriteFile(path, []byte("test-key"), 0o600); err != nil {
+		t.Fatal(err)
+	}
+	return func() { os.Remove(path) }
+}
+
+// validateConfigTestCase defines a test case for validateConfig.
+type validateConfigTestCase struct {
+	name        string
+	config      *RenewalConfig
+	expectError bool
+	errorMsg    string
+}
+
+// runValidateConfigTest runs a single validateConfig test case.
+func runValidateConfigTest(t *testing.T, tt validateConfigTestCase) {
+	cleanup := setupSSHKeyForTest(t, tt.config.ControlPlane.SSHKey)
+	defer cleanup()
+
+	err := validateConfig(tt.config)
+	if tt.expectError && err == nil {
+		t.Error("expected error but got none")
+	}
+	if !tt.expectError && err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if tt.expectError && err != nil && tt.errorMsg != "" && !contains(err.Error(), tt.errorMsg) {
+		t.Errorf("expected error message to contain %q, got %q", tt.errorMsg, err.Error())
+	}
+}
+
+// TestValidateConfig tests the validateConfig function directly.
+func TestValidateConfig(t *testing.T) {
+	tests := []validateConfigTestCase{
+		{
+			name: "valid config",
+			config: &RenewalConfig{
+				ClusterName: "test-cluster",
+				ControlPlane: NodeConfig{
+					Nodes:   []string{"192.168.1.10"},
+					OS:      "ubuntu",
+					SSHKey:  "/tmp/test-key",
+					SSHUser: "ec2-user",
+				},
+			},
+			expectError: false,
+		},
+		{
+			name: "missing cluster name",
+			config: &RenewalConfig{
+				ControlPlane: NodeConfig{
+					Nodes:   []string{"192.168.1.10"},
+					OS:      "ubuntu",
+					SSHKey:  "/tmp/test-key",
+					SSHUser: "ec2-user",
+				},
+			},
+			expectError: true,
+			errorMsg:    "cluster name is required",
+		},
+		{
+			name: "missing control plane nodes",
+			config: &RenewalConfig{
+				ClusterName: "test-cluster",
+				ControlPlane: NodeConfig{
+					OS:      "ubuntu",
+					SSHKey:  "/tmp/test-key",
+					SSHUser: "ec2-user",
+				},
+			},
+			expectError: true,
+			errorMsg:    "at least one node is required",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			runValidateConfigTest(t, tt)
```

Comment:
instead of defining a function, can we write the code here it self? Also, can we create SSH key once only for the scope of this test?